internal/provider/resource_workspace_current_state.go

```diff
@@ -0,0 +1,313 @@
+// The workspace path, module source, and module version uniquely identify this workspace_current_state.
+type WorkspaceCurrentStateModel struct {
+	WorkspacePath types.String `tfsdk:"workspace_path"`
+	ModuleSource  types.String `tfsdk:"module_source"`
+	ModuleVersion types.String `tfsdk:"module_version"`
+	Teardown      types.Bool   `tfsdk:"teardown"`
+}
+
+// Set the Teardown field's default value of false.
+func (m *WorkspaceCurrentStateModel) setDefaultTeardown() {
+	if m.Teardown.IsNull() || m.Teardown.IsUnknown() {
+		m.Teardown = types.BoolValue(false)
+	}
+}
+
+// Ensure provider defined types fully satisfy framework interfaces
+var (
+	_ resource.Resource                = (*workspaceCurrentStateResource)(nil)
+	_ resource.ResourceWithConfigure   = (*workspaceCurrentStateResource)(nil)
+	_ resource.ResourceWithImportState = (*workspaceCurrentStateResource)(nil)
+)
+
+// NewWorkspaceCurrentStateResource is a helper function to simplify the provider implementation.
+func NewWorkspaceCurrentStateResource() resource.Resource {
+	return &workspaceCurrentStateResource{}
+}
+
+type workspaceCurrentStateResource struct {
+	client *tharsis.Client
+}
+
+// Metadata returns the full name of the resource, including prefix, underscore, instance name.
+func (t *workspaceCurrentStateResource) Metadata(ctx context.Context,
+	req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "tharsis_workspace_current_state"
+}
+
+func (t *workspaceCurrentStateResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	description := "Defines and manages a workspace current state."
+
+	resp.Schema = schema.Schema{
+		Version:             1,
+		MarkdownDescription: description,
+		Description:         description,
+		Attributes: map[string]schema.Attribute{
+			"workspace_path": schema.StringAttribute{
+				MarkdownDescription: "The full path of the workspace.",
+				Description:         "The full path of the workspace.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"module_source": schema.StringAttribute{
+				MarkdownDescription: "The source of the module, including the API hostname.",
+				Description:         "The source of the module, including the API hostname.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"module_version": schema.StringAttribute{
+				MarkdownDescription: "The version identifier of the module.",
+				Description:         "The version identifier of the module.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"teardown": schema.BoolAttribute{
```

Comment:
Teardown shouldn't be an option. That is what should happen on a destroy. Otherwise, destroy would be the opposite and apply on a destroy.